src/cpp/session/modules/rmarkdown/NotebookData.cpp

```diff
@@ -0,0 +1,115 @@
+/*
+ * NotebookData.cpp
+ *
+ * Copyright (C) 2009-16 by RStudio, Inc.
+ *
+ * Unless you have received this program directly from RStudio pursuant
+ * to the terms of a commercial license agreement with RStudio, then
+ * this program is licensed to you under the terms of version 3 of the
+ * GNU Affero General Public License. This program is distributed WITHOUT
+ * ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING THOSE OF NON-INFRINGEMENT,
+ * MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
+ * AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
+ *
+ */
+
+#include "SessionRmdNotebook.hpp"
+#include "NotebookData.hpp"
+#include "NotebookOutput.hpp"
+
+#include <iostream>
+
+#include <boost/foreach.hpp>
+#include <boost/format.hpp>
+
+#include <r/RExec.hpp>
+#include <r/RRoutines.hpp>
+
+#include <core/Exec.hpp>
+
+#include <session/SessionModuleContext.hpp>
+
+using namespace rstudio::core;
+
+
+#define kNotebookDataResource "rmd_data"
+#define kNotebookDataResourceLocation "/" kNotebookDataResource "/"
+
+namespace rstudio {
+namespace session {
+namespace modules {
+namespace rmarkdown {
+namespace notebook {
+
+namespace {
+
+SEXP rs_recordData(SEXP dataFileSEXP)
+{
+   events().onDataOutput(FilePath(r::sexp::safeAsString(dataFileSEXP)), FilePath());
+   return R_NilValue;
+}
+
+void handleNotebookDataResReq(const http::Request& request,
+                              http::Response* pResponse)
+{
+   std::string resourceName = http::util::pathAfterPrefix(request, kNotebookDataResourceLocation);
+
+   std::string path;
+   Error error = r::exec::RFunction(
+      ".rs.packageFilePath",
+      "rmd/h/pagedtable-0.0.1/" + resourceName,
```

Comment:
Since the version of pagedtable-0.0.1 may change from under us I think we should implement a (private) function within the rmarkdown package that provides the script directly (we can return the string using `pResponse->setBody`)